app/src/main/java/com/nextcloud/talk/conversationlist/viewmodels/ConversationsListViewModel.kt

```diff
@@ -88,6 +94,20 @@ class ConversationsListViewModel @Inject constructor(
         repository.getRooms()
     }
 
+    fun updateRoomMessages(list: List<ConversationModel>) {
+        val current = list.associateWith { model ->
+            val unreadMessages = model.unreadMessages
+            unreadMessages
+        }
+        viewModelScope.launch(Dispatchers.IO) {
+            for ((model, unreadMessages) in current) {
+                if (unreadMessages > 0) {
+                    chatRepository.updateRoomMessages(model.internalId, unreadMessages)
+                }
+            }
+        }
+    }
+
```

Comment:
Could be simplified to this?:

fun updateRoomMessages(list: List<ConversationModel>) {
    viewModelScope.launch(Dispatchers.IO) {
        list.filter { it.unreadMessages > 0 }
            .forEach { chatRepository.updateRoomMessages(it.internalId, it.unreadMessages) }
    }
}